src/spec-common/dotfiles.ts

```diff
@@ -44,20 +44,34 @@ export async function installDotfiles(params: ResolverParameters, properties: Co
 ${createFileCommand(markerFile)} || (echo dotfiles marker found && exit 1) || exit 0
 command -v git >/dev/null 2>&1 || (echo git not found && exit 1) || exit 0
 [ -e ${targetPath} ] || ${allEnv}git clone ${repository} ${targetPath} || exit $?
-if [ -x ${installCommand} ]
+echo Setting current directory to ${targetPath}
+cd ${targetPath}
+
+if [ -f ${targetPath}/${installCommand} ]
 then
-	echo Executing script ${installCommand}
-	cd ${targetPath} && ${allEnv}${installCommand}
+	if [ ! -x ${targetPath}/${installCommand} ]
+	then
+		echo Setting '${targetPath}'/'${installCommand}' as executable
+		chmod +x ${targetPath}/${installCommand}
+	fi
+	echo Executing command '${targetPath}/${installCommand}'...\n
+	${allEnv}${targetPath}/${installCommand}
 else
-	echo Error: ${installCommand} not executable
-	exit 126
+	if [ ! -x ${installCommand} ]
```

Comment:
Add `elif [ -f "${installCommand}" ]` here and have an `else` handling when the file does not exist.